packages/runtime-dom/src/directives/vOn.ts

```diff
@@ -0,0 +1,36 @@
+const modifierGuards: Record<string, (e: Event) => void | boolean> = {
+  stop: e => e.stopPropagation(),
+  prevent: e => e.preventDefault(),
+  self: e => e.target !== e.currentTarget,
+  ctrl: e => !('ctrlKey' in e),
+  shift: e => !('shiftKey' in e),
+  alt: e => !('altKey' in e),
+  meta: e => !('metaKey' in e),
+  left: (e: MouseEvent) => 'button' in e && e.button !== 0,
+  middle: (e: MouseEvent) => 'button' in e && e.button !== 1,
+  right: (e: MouseEvent) => 'button' in e && e.button !== 2
+}
+
+// todo
+// KeyboardEvent.keyCode aliases
+const keyCodes: { [key: string]: number | Array<number> } = {
```

Comment:
Since `KeyboardEvent.keyCode` is actually deprecated, we should consider using [`KeyboardEvent.key`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) instead.

With that in mind, instead of hard coding the supported key modifiers list, we can actually do something like this:

``` js
if ('key' in event) {
  const key = event.key.toLowerCase()
  const keys = modifiers.filter(m => !modifierGuards[m])
  if (!keys.some(k => k === key)) return
}
```

So `<input @keyup.enter="onEnter">` will work like before. We can also map the 2.x key aliases to corresponding keys for backwards compat (e.g. `esc` -> `Escape`, `left` -> `LeftArrow`)